content/chapters/software-stack/lab/solution/basic-syscall/arm/Makefile

```diff
@@ -10,6 +10,9 @@ hello: hello.o
 
 hello.o: hello.s
 
+run:
```

Comment:
I'm not that fond of having a `run` rule as part of the `Makefile`. I'm mostly fond of having it part of a `README.md`. But I think we can make an exception in this case.

Reply:
Add a dependency on the `hello` executable.

Reply:
And this works out the box on my system:

```
razvan@yggdrasil:~/.../solution/basic-syscall/arm$ file hello
hello: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), statically linked, not stripped

razvan@yggdrasil:~/.../solution/basic-syscall/arm$ ./hello 
Hello, world!
```

It's because the loader is able to call the appropriate interpreter program.

Reply:
Resolved the `hello` dependency.

Out of curiosity, how did you setup your environment in order for the loader to behave in such a magic way?

Reply:
It's default. This has been the default behavior for some time.